src/sidebar/components/ThreadList.tsx

```diff
@@ -115,6 +156,25 @@ export default function ThreadList({ threads }: ThreadListProps) {
     return newAnnotations[newAnnotations.length - 1].$tag;
   })();
 
+  // Compute the heading to display above each thread.
+  //
+  // We compute the map based on the full list of threads, not just the rendered
+  // ones, so the association doesn't change while scrolling.
+  const headings = useMemo(() => {
+    let prevHeadingKey: string | null = null;
+    const headingForKey = headingMap(threads);
+
+    const headings = new Map();
+    for (const thread of threads) {
+      const key = headingKey(thread);
+      if (key && prevHeadingKey !== key) {
+        prevHeadingKey = key;
+        headings.set(thread, headingForKey.get(key) ?? 'Untitled chapter');
```

Comment:
Are we sure about the hard-coded "chapter" nomenclature?

I wonder if, later, we could introduce some design cues to different "sections" of annotation-cards that would allow us to not use any text if there isn't a `title` to show. If that makes any kind of sense. e.g. perhaps we could chunk out sub-sets of annotation cards onto different backgrounds or indentations or...other affordances that indicate they are part of a section "collection." That could obviate the need to render any text if we don't have useful metadata.

Reply:
> Are we sure about the hard-coded "chapter" nomenclature?

"section" might be better. The internal "segment" term we use might sound overly technical IMO.

> I wonder if, later, we could introduce some design cues to different "sections" of annotation-cards that would allow us to not use any text if there isn't a title to show.

Makes sense.